src/java.desktop/share/classes/javax/swing/JInternalFrame.java

```diff
@@ -1312,9 +1314,11 @@ public JDesktopPane getDesktopPane() {
     @BeanProperty(description
             = "The icon shown when this internal frame is minimized.")
     public void setDesktopIcon(JDesktopIcon d) {
-        JDesktopIcon oldValue = getDesktopIcon();
-        desktopIcon = d;
-        firePropertyChange("desktopIcon", oldValue, d);
+        if (d != null) {
+            JDesktopIcon oldValue = getDesktopIcon();
+            desktopIcon = d;
+            firePropertyChange("desktopIcon", oldValue, d);
+        }
```

Comment:
Probably NPE is better in this case? Either way the specification should say something about the new behaviour.

Reply:
Are you saying to throw NPE upfront here?
Regarding the specification, we are just preventing null as per recommendation..How should it be documented (that we are preventing null from being thrown later on)?